### PR TITLE
EPMDJ-9872 Remove offline agent

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt [hunk 1]
```diff
@@ -202,23 +206,28 @@ class AgentEndpoints(override val di: DI) : DIAware{
...
-                        call.respond(HttpStatusCode.OK, EmptyContent)
+                    val (status, message) = if (agentManager.removePreregisteredAgent(agentId)) {
+                        HttpStatusCode.OK to "Pre registered Agent '$agentId' has been completely removed"
                     } else {
-                        logger.debug { "Deleting registered Agent '$agentId' isn't supported yet" }
-                        call.respond(
-                            HttpStatusCode.BadRequest,
-                            ErrorResponse("Deleting registered Agent '$agentId' isn't supported yet")
-                        )
+                        if (buildManager.buildStatus(agentId) == BuildStatus.OFFLINE) {
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt:226]: maybe remove with all statuses?

File: core/src/main/kotlin/cache/impl/MapDBCacheService.kt [hunk 2]
```diff
@@ -67,9 +67,12 @@ class MapDBCacheService : CacheService {
...
     }
 
-    fun clear(id: Any, qualifier: Any) = dbMemory.get<HTreeMap<Any, ByteArray>>(cacheId(id, qualifier))?.clear()
+    fun clear(
+        id: Any,
+        qualifier: Any = "",
+    ) = dbMemory.getAll().filter { it.key.startsWith(cacheId(id, qualifier)) }.forEach { hTreeMap(it).clear() }
 
-    private fun cacheId(id: Any, qualifier: Any) = "$id$qualifier"
+    private fun cacheId(id: Any, qualifier: Any = "") = "$id$qualifier"
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/cache/impl/MapDBCacheService.kt]: revert

File: core/src/main/kotlin/websocket/Subscriptions.kt [hunk 3]
```diff
@@ -17,11 +17,17 @@ package com.epam.drill.admin.websocket
 
 import com.epam.drill.admin.api.websocket.*
 
+internal const val agentPrefix = "agent:"
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/websocket/Subscriptions.kt]:
```suggestion
internal const val agentPrefix = "agent::"
```
or
```suggestion
internal const val agentPrefix = "agent"
```

File: core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt [hunk 4]
```diff
@@ -202,23 +206,32 @@ class AgentEndpoints(override val di: DI) : DIAware{
                 }
             }
 
-            //TODO EPMDJ-9872 Add ability to remove offline agent
+            /**
+             * Also send action to plugin
```
Comment (Mazel-Tovr) [inline core/src/main/kotlin/endpoints/admin/AgentEndpoints.kt]:
```suggestion
             * Also you should send action to plugin
```

**Pull request merged**